### PR TITLE
Fix java 64 bit condition

### PR DESCRIPTION
**Pull request by Trial97**
Parent PR: #2069
Fixes https://discord.com/channels/1031648380885147709/1031823065937629267/1277314869149827114

### CODE REVIEW

**Review by Ryex — APPROVED** [review]

**Review by edgars-cirulis — APPROVED** [review]

**Comment by edgars-cirulis** [comment]
Oh wait, never mind. It shows a red 'X' icon initially, but it turns green whenever I change the value.

image

File: launcher/ui/widgets/JavaSettingsWidget.cpp [hunk 1]
```diff
@@ -504,7 +504,7 @@ void JavaSettingsWidget::updateThresholds()
     } else if (observedMaxMemory < observedMinMemory) {
         iconName = "status-yellow";
         m_labelMaxMemIcon->setToolTip(tr("Your maximum memory allocation is smaller than the minimum value"));
-    } else if (observedMaxMemory > 2048 && m_result.is_64bit) {
+    } else if (observedMaxMemory > 2048 && !m_result.is_64bit) {
```
Comment (TheKodeToad) [inline launcher/ui/widgets/JavaSettingsWidget.cpp:508]: IIRC this should also be 4096 for 4GB of ram (2^32/1024/1024)

Comment (Ryex) [reply launcher/ui/widgets/JavaSettingsWidget.cpp:508]: The max addressable memory on a 32bit Java is indeed 2^32 or 4096MB
However the Max *heep* size in 32bit Java is 2GB. 
This is due to single process contiguous memory constraints in 32bit x86. technically it's Os dependent
BSD / Linux in 32bit mode can go up to 3GB for a single process.

Comment (Trial97) [reply launcher/ui/widgets/JavaSettingsWidget.cpp:508]: Can we leave it as a limit of 2048?
We should not recommend that new users use Java x32 anyway.
As Ryex mentioned this is OS-dependent some will allow higher values, but 2048 is the maximum I saw working on most distros for x32bit applications.

Comment (TheKodeToad) [reply launcher/ui/widgets/JavaSettingsWidget.cpp:508]: Oh right, pretty sure -Xmx is max heap size anyway

**Review by TheKodeToad — APPROVED** [review]

**Comment by edgars-cirulis** [comment]
> Oh wait, never mind. It shows a red 'X' icon initially, but it turns green whenever I change the value.
> 
> image

It still has issue but reversed

**Pull request merged**